### ISSUE TITLE
pablo-tfg: In src/parser/lval.js check the lvalue is OK using method checkLVal

### ISSUE THREAD

**Issue by crguezl** [ULL-ESIT-PL/babel-tanhauhau#32]

**Comment by crguezl** [comment]
See https://github.com/ULL-ESIT-PL/babel-tanhauhau/blame/pablo-tfg/packages/babel-parser/src/parser/lval.js#L460-L462:

<img width="454" alt="Image" src="https://github.com/user-attachments/assets/908d1f83-5f33-459c-afef-9dcb7d2669d2" />

This bug is related to bug #31 pablo-tfg: bug empty assignment

**Comment by PSantanaGlez13** [comment]
Oh, I wrote that comment because I wasn't sure if I was being consistent with the other cases in the switch, basically because some cases where calling `checkLVal` (the same method the switch is in). I was in a rush and thought what I did was not the proper way to allow left side assignment. I believe I should add `checkLVal` to check the caller of the expression so if the caller is another `CallExpression` I can check for the number of arguments and throw if needed.

In short, I did not check the method thoroughly and did not realize the method checks the lval recursively, adding context information to the call if needed. I have also realized that I have a few TODO comments that maybe I should write as issues to explain them in more detail so they don't lead to confusions.

**Comment by crguezl** [comment]
@PSantanaGlez13 

> I believe I should add `checkLVal` to check the caller of the expression so if the caller is another `CallExpression` I can check for the number of arguments and throw if needed ...

Yes this is s.t. to do.  But first we have to fix the problem with nested calls. This issue  here is just to remember us that this is s.t. to be done in the future. The semantic problem is more general, no only `f() = 4` must throw but also f(5)() = `9`  and if we consider multiple arguments `f(2, undef, 3) = 9`  etc. *I think*[^1] it must also throw.

My feelings about that is that, to begin with,  assignable `@@` functions **must allow only one parameter** and we must forget about the multiple parameter thing for a later stage of the project.

[^1]: is just my opinion but not sure

**Comment by PSantanaGlez13** [comment]
> The semantic problem is more general, no only `f() = 4` must throw but also f(5)() = `9`...

For this case, by using the solution proposal in #31 and adding `checkLVal` as I was mentioning:
```js
      case "CallExpression":
        if (expr.arguments.length === 0) {
          this.raise(
           expr.start, 
           Errors.InvalidLhsBinding, 
           (expr?.callee?.type === "Identifier")? `"${expr.callee.name}"`: "", 
          );
        }
        this.checkLVal(
          expr.callee,
          bindingType,
          checkClashes,
          ""
        );
        break;
```

`checkLVal` is checking the expression callee (`expr.callee`), so the number of arguments is being checked for each of the nested calls (this also works if they are not nested, I think we can close #31).

> ...and if we consider multiple arguments `f(2, undef, 3) = 9` etc. I think it must also throw.
> My feelings about that is that, to begin with, assignable `@@` functions **must allow only one parameter** and we must forget about the multiple parameter thing for a later stage of the project.

I am considering it. There is a **lot** of cases that have to be covered and maybe making sure the basics (for example, nested calls) are working properly before adding multiple arguments to the mix is a more appropiate approach.

**Comment by crguezl** [comment]
@PSantanaGlez13 

Yes, it works:

Here is the test:

```js 
➜  not-working git:(main) ✗ cat nested-empty-assignment.js
function @@ foo(a) {
  return function @@ bar() {
    return a + 2;
  }
}

foo(6)() = "Another value";
console.log(foo(2)());
console.log(foo(6)()); 
```

And here is the output when adding your code:

```                                                                                                                     
➜  not-working git:(main) ✗ npx babel nested-empty-assignment.js 
SyntaxError: /Users/casianorodriguezleon/campus-virtual/2324/research/parser-left-side-crguezl/examples/not-working/nested-empty-assignment.js: Binding invalid left-hand side in  (7:0)

  5 | }
  6 |
> 7 | foo(6)() = "Another value";
    | ^
  8 | console.log(foo(2)());
  9 | console.log(foo(6)()); 
```

See folder https://github.com/ULL-ESIT-PL-2425/parser-left-side-crguezl/tree/main/examples/
> I am considering it. There is a **lot** of cases that have to be covered and maybe making sure the basics (for example, nested calls) are working properly before adding multiple arguments to the mix is a more appropriate approach.

Yes!  priority to check that nested calls works correctly.  My vote is against supporting multiple arguments in assignable functions.
